sdk/state/integration/helpers_test.go

```diff
@@ -93,63 +127,127 @@ func initAccounts(t *testing.T) (initiator Participant, responder Participant) {
-			},
-		},
-		RemoteEscrowAccount: &state.EscrowAccount{
-			Address:        responder.Escrow.FromAddress(),
-			SequenceNumber: responder.EscrowSequenceNumber,
-			Balances: []state.Amount{
-				{Asset: state.NativeAsset{}, Amount: responder.Contribution},
-			},
-		},
-		LocalSigner:  initiator.KP,
-		RemoteSigner: responder.KP.FromAddress(),
+		LocalEscrowAccount:         &initiatorEscrowAccount,
+		RemoteEscrowAccount:        &responderEscrowAccount,
+		LocalSigner:                initiator.KP,
+		RemoteSigner:               responder.KP.FromAddress(),
 	})
 	responderChannel = state.NewChannel(state.Config{
 		NetworkPassphrase:          networkPassphrase,
 		ObservationPeriodTime:      observationPeriodTime,
 		ObservationPeriodLedgerGap: observationPeriodLedgerGap,
 		Initiator:                  false,
-		LocalEscrowAccount: &state.EscrowAccount{
-			Address:        responder.Escrow.FromAddress(),
-			SequenceNumber: responder.EscrowSequenceNumber,
-			Balances: []state.Amount{
-				{Asset: state.NativeAsset{}, Amount: responder.Contribution},
+		LocalEscrowAccount:         &responderEscrowAccount,
+		RemoteEscrowAccount:        &initiatorEscrowAccount,
+		LocalSigner:                responder.KP,
+		RemoteSigner:               initiator.KP.FromAddress(),
+	})
+	return initiatorChannel, responderChannel
+}
+
+func initAsset(t *testing.T, client horizonclient.ClientInterface) (txnbuild.Asset, *keypair.Full) {
+	issuerKP := keypair.MustRandom()
+	distributorKP := keypair.MustRandom()
+
+	err := retry(2, func() error { return createAccount(client, issuerKP.FromAddress(), 1_000_0000000) })
+	require.NoError(t, err)
+	err = retry(2, func() error { return createAccount(client, distributorKP.FromAddress(), 1_000_0000000) })
+	require.NoError(t, err)
+
+	issuer, err := client.AccountDetail(horizonclient.AccountRequest{AccountID: issuerKP.Address()})
+	require.NoError(t, err)
+	distributor, err := client.AccountDetail(horizonclient.AccountRequest{AccountID: distributorKP.Address()})
+	require.NoError(t, err)
+
+	abcdAsset := txnbuild.CreditAsset{Code: "ABCD", Issuer: issuerKP.Address()}
+
+	tx, err := txnbuild.NewTransaction(
+		txnbuild.TransactionParams{
+			SourceAccount:        &distributor,
+			IncrementSequenceNum: true,
+			BaseFee:              txnbuild.MinBaseFee,
+			Timebounds:           txnbuild.NewInfiniteTimeout(),
+			Operations: []txnbuild.Operation{
+				&txnbuild.ChangeTrust{
+					Line:  abcdAsset,
+					Limit: "5000",
+				},
 			},
 		},
-		RemoteEscrowAccount: &state.EscrowAccount{
-			Address:        initiator.Escrow.FromAddress(),
-			SequenceNumber: initiator.EscrowSequenceNumber,
-			Balances: []state.Amount{
-				{Asset: state.NativeAsset{}, Amount: initiator.Contribution},
+	)
+	tx, err = tx.Sign(networkPassphrase, distributorKP)
+	require.NoError(t, err)
+	_, err = client.SubmitTransaction(tx)
+	require.NoError(t, err)
+
+	tx, err = txnbuild.NewTransaction(txnbuild.TransactionParams{
```

Comment:
can combine the two actions into a single transaction rather than needing two stellar transactions

Reply:
good point 👍 
https://github.com/stellar/experimental-payment-channels/pull/86/commits/f0148b40ba5ac56100869c32890fc1f30ca5e49f